mempool/src/shared_mempool/types.rs

```diff
@@ -69,6 +73,42 @@ pub(crate) fn notify_subscribers(
     }
 }
 
+/// represents `k-policy` to ensure reliable txn delivery:
+/// - removes txn from mempool if at least `k` peers have successfully ACK'ed for it
+#[derive(Clone)]
+pub(crate) struct AckPolicy {
+    // tracks how many ACKs have been received for a pending broadcasted txn
+    // (k, v) = (txn timeline_id, # ACKs received)
+    pub ack_counts: HashMap<u64, usize>,
```

Comment:
a bit confused about how it works. It looks like you keep entry for each timeline_id that was broadcasted and only clear it up once k acks were received
couple of questions:
1. let's say transactions were committed and removed from core mempool. Won't it leave dangling pointer in this map? Cause in task you only iterate through entries that are still present in mempool
2. how works the VFN case? (e.g. where we have < k primaries, but need to enforce at least k for secondaries)